---
author: nithyasarabudla
output:
  html_document:
    toc: true
    toc_depth: 4
    toc_float: true
    dev: 'svg'
  md_document:
    variant: gfm
bibliography: bibliography.ris
---
## Methods
Variant calling is the identification of genomic differences between individuals of the same species (germline) or differences between tissues in a single individual (somatic). The basic process for finding germline variation is to align NGS reads from an individual against the reference genome for the species producing a bam alignment file then to analyze the alignments using a variant calling program like the Genome Analysis Tool Kit (GATK)[@GenomeAnalysisToolkit] or DeepVariant. The output is generally stored as a Variant Call Format, or vcf file. The process is basically the same for finding somatic variants, the key difference being that you are looking for genetic variation between the tissues of a single individual. A very important use for somatic variant calling is identifying mutations in cancer cells within a tumor.
# Reference genome retrieval (human, release 27) via the Gencode FTP server
# getGenome.sh
1.The getGenome.sh script allows us to retrieve the GRCh38 reference genome.
2.The script saves the GRCh38 reference genome as GRCh38_reference.fa.gz after downloading it in FASTA format from the Gencode FTP server. 
3.The file is downloaded using the "wget" command. 
4.The output file name is specified by the -O option. The “wget” command redirects the standard output and standard error streams to two log files, getGenome.log and getGenome.err, respectively.

# Retrieve the NGS reads available in the SRA under accession SRR6808334 
# getReads.sh
1.The getReads.sh script allows us to retrieve NGS reads from the NA12878 reference sample. 
2.The fastq-dump command is used to download the reads in FASTQ format from the Sequence Read Archive (SRA) database. 
3.The --split-files option is used to split the paired-end reads into two separate files. The script redirects the standard output and standard error streams of the fastq-dump command to two log files, getReads.log and getReads.err.

# Quality trim the reads using Trimmomatic
# trimReads.sh
1.Trimmomatic software is used to trim and filter Next-Generation Sequencing (NGS) reads.[@Trimmomatic]
2.The paired-end input reads are indicated by "PE”
3.-phred33: indicates that the input quality scores are in the Phred+33 format.
4.SRR6808334_1.fastq and SRR6808334_2.fastq indicate the file name of the input forward reads and input reverse reads.
5.SRR6808334_1.paired.fastq and SRR6808334_1.unpaired.fastq specify the file name for the output forward paired-end reads and unpaired reads.
6.SRR6808334_2.paired.fastq and SRR6808334_2.unpaired.fastq specify the file name for the output reverse paired-end reads and reverse unpaired reads.
7.HEADCROP:0 specifies no bases should be removed from the starting of each read.
8.LEADING:20 TRAILING:20: specifies the minimum quality score required for the leading and trailing bases of each read.
9.SLIDINGWINDOW:4:30: specifies the window size and average quality score required for sliding window trimming.
10.MINLEN:36: specifies the minimum length of the trimmed reads.
11.The script creates two log files called trimReads.log and trimReads.err from the standard output and standard error streams of the Trimmomatic command.

# Index the genome for use by BWA 
# indexGenome.sh
1.bwa index command is used to create an index for the reference genome GRCh38_reference.fa.
2.IndexGenome.log and IndexGenome.err are two log files that contain the standard output and standard error streams of the bwa index command.[@ShortReadAlignment]

# Align the reads using bwa mem
# alignReads.sh
1.bwa mem command aligns the  paired-end reads from the fastq files SRR6808334_1.fastq and SRR6808334_2.fastq to the reference genome.
2.-t 8 indicates the number of threads to use for alignment.
3.BWA-MEM is instructed to use paired-end reads when given the -p option.
4.The SRR6808334.sam and alignReads.err log files contain the standard output and standard error streams of the bwa mem command.

# Sort the file created by bwa mem to a sorted bam 
# sort.sh
1.samtools sorts the SAM file generated by alignReads.sh and converts it to a BAM file. 
2.samtools sort command to sort the SAM file SRR6808334.sam and save the sorted BAM file as SRR6808334.bam. 
3.-@ 8 option indicates the number of threads to use.
4.Two log files, sort.log and sort.err, are used to store the standard output and standard error streams of the samtools sort command.

# Index reads
#indexReads.sh
1."samtools index" command to create an index file for a BAM file called "SRR6808334.bam".
2.The script redirects standard output (stdout) to a log file called "index.log" using the ">" symbol and redirects standard error (stderr) to a separate file called "index.err" using the "2>" symbol

# Produce a VCF file using DeepVariant
# runDeepVariant.sh
1.Variables are created to set up the file paths, location of the input files, output directory, and reference genome.
2.By using the "mkdir -p" command, the necessary directories are created.
3.The script checks if Docker is already installed. If it isn't, it installs Docker using the official Docker installation instructions for Ubuntu.
4.The script then copies the input files, such as the BAM file and reference genome, to the input directory.
5.The "docker pull" command pulls the DeepVariant Docker image.
6.Finally, the script runs DeepVariant using the "docker run" command. The output VCF and GVCF files[@IndelVariantCaller] will be written to the output directory.

## References
